public/app/types/explore.ts

```diff
@@ -233,3 +233,13 @@ export type RichHistoryQuery = {
   sessionName: string;
   timeRange?: string;
 };
+
+export interface DecoratedPanelData extends PanelData {
+  graphFrames: DataFrame[];
+  tableFrames: DataFrame[];
+  logsFrames: DataFrame[];
+  traceFrames: DataFrame[];
+  graphResult: GraphSeriesXY[] | null;
+  tableResult: DataFrame | null;
+  logsResult: LogsModel | null;
+}
```

Comment:
I'm not sure about the **name** of the interface. I would expect a type named `DecoratedPanelData` to be a generic one, something like this:

```
type DecoratedPanelData<T extends {}> = PanelData & T;
```

And in this case I guess the better name would just be `ExplotePanelData` (that  could be a `DecoratedPanelData<{...explore props}>`)

Reply:
Great finding, I like that name better!